src/chunk_constraint.c

```diff
@@ -674,15 +681,21 @@ chunk_constraint_drop_constraint(TupleInfo *ti)
 		heap_getattr(ti->tuple, Anum_chunk_constraint_constraint_name, ti->desc, &isnull);
 	int32 chunk_id =
 		DatumGetInt32(heap_getattr(ti->tuple, Anum_chunk_constraint_chunk_id, ti->desc, &isnull));
-	Chunk *chunk = ts_chunk_get_by_id(chunk_id, 0, true);
-	ObjectAddress constrobj = {
-		.classId = ConstraintRelationId,
-		.objectId =
-			get_relation_constraint_oid(chunk->table_id, NameStr(*DatumGetName(constrname)), true),
-	};
+	/* Get the chunk relid. Note that, at this point, the chunk table can be
+	 * deleted already. */
+	Oid chunk_relid = ts_chunk_get_relid(chunk_id, true);
 
-	if (OidIsValid(constrobj.objectId))
-		performDeletion(&constrobj, DROP_RESTRICT, 0);
+	if (OidIsValid(chunk_relid))
```

Comment:
The other condition should not happen. Either remove the condition or replace with assert.

Reply:
I am not sure I follow. What other condition are you referring to?

Reply:
I mean the `else` case.

Reply:
I.e., `OidIsValid(chunk_relid)`is always true.

Reply:
Actually I am wrong the function call specified `missing_ok = true`.

Reply:
It was `false` before.
I guess the change supports dropped case when no metadata should be deleted and no error should be produced.